relay-server/src/endpoints/common.rs

```diff
@@ -373,68 +357,26 @@ pub async fn handle_envelope(
     }
 }
 
-/// Creates a HttpResponse containing the textual representation of the given EventId
-pub fn create_text_event_id_response(id: Option<EventId>) -> HttpResponse {
-    // Event id is set statically in the ingest path.
-    let id = id.unwrap_or_default();
-    debug_assert!(!id.is_nil());
-
-    // the minidump client expects the response to contain an event id as a hyphenated UUID
-    // i.e. xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
-    HttpResponse::Ok()
-        .content_type("text/plain")
-        .body(id.0.as_hyphenated().to_string())
-}
+#[derive(Debug)]
+pub struct TextResponse(pub Option<EventId>);
```

Comment:
Unfortunately, we need to allow `None` here because of static typing. Our endpoints need to be defensive, which is why we're not going to error in production in case the event ID ends up missing. In tests we'd of course still know about this bug. 

I haven't found a good way to address this yet in a type-safe manner. Maybe we could start to type envelopes into event-carrying and event-less Envelopes, one of which have a guaranteed event_id. That's a much bigger change, though.